src/Lean/Elab/Deriving/DecEq.lean

```diff
@@ -169,7 +169,7 @@ def mkEnumOfNatThm (declName : Name) : MetaM Unit := do
       value, type
     }
 
-def mkDecEqEnum (declName : Name) : CommandElabM Unit := do
+def mkDecEqEnum (declName : Name) : CommandElabM Bool := do
```

Comment:
Why have you changed all the return types to `Bool`, give  that these functions always return `true`? Seems clearer to just add a `return true` at the end of `mkDecEqInstanceHandler`.

Reply:
`mkDecEq` currently returns false if the inductive type is nested, hence why I've thought it would be better to propagate the bool here, but you're right that it's not useful for `mkDecEqEnum`, I'll change that.
In practice, that boolean is only used to throw an error saying that the type isn't handled when it isn't ([here](https://github.com/leanprover/lean4/blob/4497aba1a985619023adf0e1c655e769b12735e7/src/Lean/Elab/Deriving/Basic.lean#L91)), but if that's the case, why not throw the error locally instead of propagating it that way ?

Reply:
Well, it's actually using that flag to decide to call the default handler, which currently happens to error, but might not in future. (E.g. in Lean3 the default handler unfolded the head constant and then looked for an instance.)